components/style/parallel.rs

```diff
@@ -125,6 +128,19 @@ pub fn traverse_dom<E, D>(traversal: &D,
     }
 }
 
+/// A callback to create our thread local context.  This needs to be
+/// out of line so we don't allocate stack space for the entire struct
+/// in the caller.
+#[inline(never)]
+fn create_thread_local_context<'scope, E, D>(
+    traversal: &'scope D,
+    slot: &mut RefMut<Option<D::ThreadLocalContext>>)
```

Comment:
You don't need to pass a RefMut here, because this is stack-scoped. Just pass &mut Option<...>.